File: SUAPC_2020/2A.cpp
#include <iostream>
using namespace std;

int main() {
	double d1, d2, d3; cin >> d1 >> d2 >> d3;
	double sum = (d1 + d2 + d3) / 2;

	bool impossible = false;
	double a = sum - d3;
	if (a <= 0) impossible = true;
	double b = sum - d2;
	if (b <= 0) impossible = true;
	double c = sum - d1;
	if (c <= 0) impossible = true;

	if (impossible) cout << -1 << endl;
	else {
		cout << 1 << endl;
		printf("%.1f %.1f %.1f\n", a, b, c);
	}
}

File: SUAPC_2020/2A.py
d1, d2, d3 = map(int, input().split())
sum = (d1 + d2 + d3) / 2

impossible = False
a = sum - d3
if a <= 0: impossible = True
b = sum - d2
if b <= 0: impossible = True
c = sum - d1
if c <= 0: impossible = True

if impossible: print(-1)
else:
    print(1)
    print("%.1f %.1f %.1f" % (a, b, c))
